### PR TITLE
Feature 79/manage houses

### PR DESCRIPTION
**Pull request by rostovskiy88**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by akononenko2375 — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -23,10 +23,12 @@
     "serve-static": "^1.14.1",
     "vue": "^3.0.0",
     "vue-router": "^4.0.0-0",
+    "vuelidate": "^0.7.6",
     "vuex": "^4.0.0-0"
   },
   "devDependencies": {
     "@types/jest": "^24.0.19",
+    "@types/vuelidate": "^0.7.15",
```
Comment (aleonovdementiev) [inline package.json]: Why do we need this dependencies if we have similar in 16 and 17 lines

**Comment by aleonovdementiev** [comment]
Please, revert package.json and package-lock.json, because packages, you've added, already exists in project.

File: src/store/houses/actions.ts [hunk 2]
```diff
@@ -12,4 +13,29 @@ export const actions: ActionTree<HousesStateInterface, RootStateInterface> & Act
...
     commit(HousesMutationsEnum.SET_MODAL_DISPLAY, payload);
   },
+  [HousesActionsEnum.EDIT_HOUSE]: async ({ commit }, payload) => {
+    try {
+      const payloadtoSend: HouseDTOModel = {
+        quantity_flat: payload.quantity_flat,
+        house_area: payload.house_area,
+        adjoining_area: payload.adjoining_area,
+        address: payload.address,
+      };
```
Comment (aleonovdementiev) [inline src/store/houses/actions.ts]: Please, extract model from actions.ts file to the separate file

File: src/store/houses/actions.ts [hunk 3]
```diff
@@ -12,4 +13,29 @@ export const actions: ActionTree<HousesStateInterface, RootStateInterface> & Act
...
+      await HTTP.put(`/cooperations/1/houses/${payload.id}`, payloadtoSend).then((r) => {
```
Comment (aleonovdementiev) [inline src/store/houses/actions.ts]: Why do you use hardcoded id here?

File: src/store/houses/actions.ts [hunk 4]
```diff
@@ -12,4 +13,29 @@ export const actions: ActionTree<HousesStateInterface, RootStateInterface> & Act
...
+        console.log(r.data);
+        commit(HousesMutationsEnum.EDIT_HOUSE, r.data);
+      });
+    } catch (err: any) {
+      console.log('error EDIT_HOUSE', err);
+    }
+  },
+  [HousesActionsEnum.DELETE_HOUSE]: async ({ commit }, payload) => {
+    try {
+      await HTTP.delete(`/cooperations/1/houses/${payload.id}`).then(() => {
```
Comment (aleonovdementiev) [inline src/store/houses/actions.ts]: And here?

File: src/store/houses/mutations.ts [hunk 5]
```diff
@@ -8,4 +8,22 @@ export const mutations: MutationTree<HousesStateInterface> & Mutations = {
...
   },
+  [HousesMutationsEnum.EDIT_HOUSE]: (state, payload) => {
+    state.houses?.forEach((el: HouseInterface) => {
+      if (el.id === payload.id) {
+        el.quantity_flat = payload.quantity_flat;
+        el.house_area = payload.house_area;
+        el.adjoining_area = payload.adjoining_area;
+        el.address = payload.address;
+      }
+    });
```
Comment (aleonovdementiev) [inline src/store/houses/mutations.ts:19]: You can extract this logic to the method in the model and then just use this method here

File: src/store/houses/mutations.ts [hunk 6]
```diff
@@ -8,4 +8,22 @@ export const mutations: MutationTree<HousesStateInterface> & Mutations = {
...
+  },
+  [HousesMutationsEnum.DELETE_HOUSE]: (state, payload) => {
+    const updateCont = state.houses?.filter((el: HouseInterface) => el.id !== payload.id);
+
+    console.log('state before MUT', state.houses);
+    console.log('updateCont', updateCont);
+    state.houses = updateCont;
+    console.log('state after MUT', state.houses);
```
Comment (aleonovdementiev) [inline src/store/houses/mutations.ts]: Do you need all this console.logs here?

File: src/views/CooperationInfo.vue [hunk 7]
```diff
@@ -187,30 +193,158 @@
...
+                    @blur="v$.house.adjoining_area.$touch"
+                  />
+                  <small v-if="v$.house.adjoining_area.$error" class="p-error">{{
+                    v$.house.adjoining_area.$errors[0].$message
+                  }}</small>
+                </p>
+                <p>
+                  <label class="dialog-item" for="coopEmail">
+                    <pre>Адреса:</pre>
+                    <pre>     Регіон :</pre>
```
Comment (aleonovdementiev) [inline src/views/CooperationInfo.vue]: It's better to make this whitespaces in styles, but not with spaces in the element

File: src/views/CooperationInfo.vue [hunk 8]
```diff
@@ -254,25 +403,29 @@ export default defineComponent({
...
   },
   data() {
     return {
-      houseActions: [
-        {
-          label: 'Видалити',
-          icon: 'pi pi-times',
-          command: () => {
-            this.confirmDeleteHouse();
+      houseActions: (house: HouseInterface) => {
```
Comment (aleonovdementiev) [inline src/views/CooperationInfo.vue:400]: Why do you need house argument if you never use it in function. By the way, you can put an array in houseActions, not the function that returns an array

File: src/views/CooperationInfo.vue [hunk 9]
```diff
@@ -361,6 +564,8 @@ export default defineComponent({
       this.closeCooperationModal();
     },
     editHouseInfo(house: HouseInterface) {
+      console.log(house);
```
Comment (aleonovdementiev) [inline src/views/CooperationInfo.vue]: remove this console.log if you don't need it.

File: src/views/CooperationInfo.vue [hunk 10]
```diff
@@ -371,10 +576,32 @@ export default defineComponent({
       this.$store.dispatch(`${StoreModuleEnum.housesStore}/EDIT_HOUSE`, payload);
       this.closeEditHouseModal();
     },
-    toggle(event: Event) {
+    toggle(event: any, slotProps: any) {
+      console.log(slotProps.quantity_flat);
```
Comment (aleonovdementiev) [inline src/views/CooperationInfo.vue]: Same here

File: src/views/CooperationInfo.vue [hunk 11]
```diff
@@ -371,10 +576,32 @@ export default defineComponent({
...
+      this.house = slotProps;
+      (this.$refs.menu as any).toggle(event);
```
Comment (aleonovdementiev) [inline src/views/CooperationInfo.vue:590]: you can call toggleConfirm here to prevent code duplicate.

File: src/store/houses/mutations.ts (hunk 6 above)
Comment (rostovskiy88) [reply src/store/houses/mutations.ts]: removed console.logs

File: src/views/CooperationInfo.vue (hunk 9 above)
Comment (rostovskiy88) [reply src/views/CooperationInfo.vue]: removed

File: src/views/CooperationInfo.vue (hunk 8 above)
Comment (rostovskiy88) [reply src/views/CooperationInfo.vue:400]: I need "house" argument for showSuccessEdit(house) inside houseActions, so i guess I will keep function with argument

File: src/views/CooperationInfo.vue (hunk 10 above)
Comment (rostovskiy88) [reply src/views/CooperationInfo.vue]: fixed

File: src/views/CooperationInfo.vue (hunk 11 above)
Comment (rostovskiy88) [reply src/views/CooperationInfo.vue:590]: removed toggleConfirm since i dont use it anymore

File: src/store/houses/mutations.ts (hunk 5 above)
Comment (rostovskiy88) [reply src/store/houses/mutations.ts:19]: May i suggest to refactor this part on next sprint? I still nedd a little bit more time to figure out how models work in TS, so i can do it properly

File: src/store/houses/actions.ts (hunk 4 above)
Comment (rostovskiy88) [reply src/store/houses/actions.ts]: fixed

File: src/views/CooperationInfo.vue (hunk 7 above)
Comment (rostovskiy88) [reply src/views/CooperationInfo.vue]: removed spaces and added styles for proper layout

File: src/store/houses/actions.ts (hunk 2 above)
Comment (rostovskiy88) [reply src/store/houses/actions.ts]: fixed

File: src/store/houses/actions.ts (hunk 3 above)
Comment (rostovskiy88) [reply src/store/houses/actions.ts]: I will fix this hardcoded id eventually when we will be able to choose from different cooperations.

File: package.json (hunk 1 above)
Comment (rostovskiy88) [reply package.json]: fixed

**Comment by rostovskiy88** [comment]
> Please, revert package.json and package-lock.json, because packages, you've added, already exists in project.

fixed

File: src/store/houses/actions.ts [hunk 12]
```diff
@@ -12,6 +12,26 @@ export const actions: ActionTree<HousesStateInterface, RootStateInterface> & Act
   [HousesActionsEnum.SET_MODAL_DISPLAY]: ({ commit }, payload) => {
     commit(HousesMutationsEnum.SET_MODAL_DISPLAY, payload);
   },
+
+  [HousesActionsEnum.EDIT_HOUSE]: async ({ commit }, payload) => {
+    try {
+      await HTTP.put(`/cooperations/1/houses/${payload.id}`, payload).then((r) => {
+        console.log(r.data);
```
Comment (aleonovdementiev) [inline src/store/houses/actions.ts]: Remove this console.log please.

File: src/store/houses/actions.ts (hunk 3 above)
Comment (aleonovdementiev) [reply src/store/houses/actions.ts]: Why can't you do it now?

File: src/store/houses/actions.ts (hunk 4 above)
Comment (aleonovdementiev) [reply src/store/houses/actions.ts]: Not fixed - hardcoded ID is still there)

File: src/store/houses/mutations.ts (hunk 5 above)
Comment (aleonovdementiev) [reply src/store/houses/mutations.ts:19]: Yep, okay, but remind me please about this item.

File: src/views/CooperationInfo.vue (hunk 8 above)
Comment (aleonovdementiev) [reply src/views/CooperationInfo.vue:400]: Yep, i missed that line of code, my bad

File: src/store/houses/actions.ts (hunk 3 above)
Comment (rostovskiy88) [reply src/store/houses/actions.ts]: removed hardcoded cooperation ID

File: src/store/houses/actions.ts (hunk 4 above)
Comment (rostovskiy88) [reply src/store/houses/actions.ts]: removed hardcoded cooperation ID

File: src/store/houses/actions.ts (hunk 12 above)
Comment (rostovskiy88) [reply src/store/houses/actions.ts]: removed conosle.log

**Review by aleonovdementiev — APPROVED** [review]

**Review by akononenko2375 — APPROVED** [review]

**Pull request merged**